tests/features/S2N_KTLS_KEYUPDATE_SUPPORTED.c

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/* Gate kTLS support only to Linux. Add other platforms once they have been tested. */
+#if defined(__linux__)
+    #include <linux/snmp.h>
```

Comment:
Simple Network Management Protocol? confused?

Reply:
Yeah you're right, it would make more sense to use the [newly defined constant in tls.h](https://lore.kernel.org/netdev/2f5f05db51d624c47fe92e27f02c922dc7d9c035.1734013874.git.sd@queasysnail.net/)

Reply:
Actually, I'm not sure I can do that. [linux/snmp.h is in a uapi directory](https://github.com/torvalds/linux/blob/master/include/uapi/linux/snmp.h), [whereas tls.h is technically not](https://github.com/torvalds/linux/blob/master/include/net/tls.h)? I am not sure if include/net/tls.h is meant to be used or if it's like an internal header file.
I'll try to see if I can include the net/tls.h in my feature flag, but not sure if it is supposed to be used.

Reply:
Are the new constants not defined in the same place as the old ones? I don't really remember why / how, but this is the current feature probe: https://github.com/aws/s2n-tls/blob/main/tests/features/S2N_KTLS_SUPPORTED.c#L17-L19

Reply:
To be precise, here is the entire key update patch: https://lore.kernel.org/netdev/cover.1734013874.git.sd@queasysnail.net/. Notice how there are only two header files touched in the include directory:
include/net/tls.h
include/uapi/linux/snmp.h
I know how to reference snmp.h, but I'm not quite sure if I'm supposed to be accessing net/tls.h. I need to do a little more testing to see if this is accessible.
And no, actually the new constants are not defined in the same place as the old ones we used in the S2N_KTLS_SUPPORTED flag.